bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/discovery/ElroConnectsBridgeDiscoveryService.java

```diff
@@ -142,8 +142,8 @@ public void deactivate() {
     @Override
     public void setThingHandler(@Nullable ThingHandler handler) {
         ElroConnectsAccountHandler account = null;
-        if (handler instanceof ElroConnectsAccountHandler) {
-            account = (ElroConnectsAccountHandler) handler;
+        if (handler instanceof ElroConnectsAccountHandler ahandler) {
+            account = ahandler;
```

Comment:
This could be simplified further by removing line 144 and lines 150-152:
```suggestion
        if (handler instanceof ElroConnectsAccountHandler accountHandler) {
            this.accountHandler = accountHandler;
            accountHandler.setDiscoveryService(this);
```

Reply:
`this.` should be removed from last line like in the suggested code to avoid null annotation issue (other thread could have set `this.accountHandler` to null in meantime).